### ISSUE TITLE
Feature request: make avilable __FUNC_NAME__ in stdlib.ml

### ISSUE THREAD

**Issue by Et7f3** [ocaml/ocaml#8824]
```
# let func_name () () =
    print_endline __FUNC_NAME__;;
val func_name : unit -> unit -> unit = <fun>
# func_name () ();;
func_name
- : unit = ()
```
I don't know which `__FUNC_NAME__` will be useful in case of partial application. Maybe `__FUNC_NAMES__` it will be a concatenation of func name, list of func name or the last name.

I don't know which name will be useful for anonymous function.

**Comment by dra27** [comment]
I don't think there's anything about this which couldn't be solved with a ppx? It's not unrelated to the `#trace` directive in the toplevel.

**Comment by Et7f3** [comment]
Oh I have forgotten `#trace`. Is there a similar mechanism for executable? Something inside OCAMLRUNPARAM for instance to activate trace ?

**Comment by nojb** [comment]
Fixed by #9554
